chrome/content/coniks/vrf.js

```diff
@@ -0,0 +1,32 @@
+this.EXPORTED_SYMBOLS = ["VRF"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource://gre/modules/ctypes.jsm");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/coniks/libconiks.js");
+Cu.import("chrome://otr/content/coniks/util.js");
+
+var VRF = {
+  /**
+   * @param {Array} pk - VRF public key.
+   * @param {string} m - The message.
+   * @param {Array} index - VRF index.
+   * @param {Array} proof - VRF proof.
+   */
+  verify: function(pk, m, index, proof) {
+    let msg = util.stringToArray(m);
+
+    let pkPtr = ctypes.unsigned_char.array()(pk);
+    let msgPtr = ctypes.unsigned_char.array()(m);
+    let indexPtr = ctypes.unsigned_char.array()(index);
+    let proofPtr = ctypes.unsigned_char.array()(proof);
+    if (libCONIKS.cgoVerifyVrf(pkPtr, pk.length,
```

Comment:
Hmm, could you please elaborate on this? Why do we want to perform that comparison?